lsp/server/src/utils/gqlUtils.ts

```diff
@@ -0,0 +1,420 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+import type { ASTNode, FieldNode, OperationDefinitionNode } from 'graphql';
+
+import { visit, Kind } from 'graphql';
+import { ObjectInfoRepresentation } from '../types';
+
+/**
+ * Represent an entity in graphql query.
+ */
+export interface EntityNode {
+    // Graphql AST node
+    node: FieldNode;
+    // Entity name, like `Account`
+    name: string | undefined;
+    // Total size of immediate children fields
+    size: number | undefined;
```

Comment:
Suppose there was no children. Then, shouldn't `size` suffice to be only `number`? Can it also be `undefined`?

Reply:
'undefined' means it is unknown yet. When 'EntityNode' is constructed in 'EntityTree', it is possible that both 'name' and 'size' are unknown. These properties will become available when traversal happens and objectInfo is fetching .